### PR TITLE
Linting fixes for the provisioning package

### PR DESCRIPTION
**Pull request by bergquist**
ohhh the joy of cleaning up code

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -832,6 +832,8 @@ jobs:
               ./pkg/services/alerting/... \
               ./pkg/services/provisioning/datasources/... \
               ./pkg/services/provisioning/dashboards/... \
+              ./pkg/services/provisioning/notifers/... \
```
Comment (aknuds1) [inline .circleci/config.yml:835]:
```suggestion
              ./pkg/services/provisioning/notifiers/... \
```

File: pkg/services/provisioning/values/values.go [hunk 2]
```diff
@@ -19,11 +19,14 @@ import (
 	"github.com/grafana/grafana/pkg/util/errutil"
 )
 
+// IntValue represents a string value in a yaml
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:22]: I think it's best to stick with the acronym YAML, lest we get PRs to change it
```suggestion
// IntValue represents a string value in a YAML
```

File: pkg/services/provisioning/values/values.go [hunk 3]
```diff
@@ -19,11 +19,14 @@ import (
...
+// config that can be overridden by environment variables
 type IntValue struct {
 	value int
 	Raw   string
 }
 
+// UnmarshalYAML converts yaml into an *IntValue
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:29]:
```suggestion
// UnmarshalYAML converts yaml into an *IntValue
```

File: pkg/services/provisioning/values/values.go [hunk 4]
```diff
@@ -38,15 +41,19 @@ func (val *IntValue) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return errutil.Wrap("cannot convert value int", err)
 }
 
+// Value returns the wrapped int value
 func (val *IntValue) Value() int {
 	return val.value
 }
 
+// Int64Value represents a string value in a yaml
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:49]:
```suggestion
// Int64Value represents a string value in a YAML
```

File: pkg/services/provisioning/values/values.go [hunk 5]
```diff
@@ -38,15 +41,19 @@ func (val *IntValue) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
+// config that can be overridden by environment variables
 type Int64Value struct {
 	value int64
 	Raw   string
 }
 
+// UnmarshalYAML converts yaml into an *Int64Value
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:56]:
```suggestion
// UnmarshalYAML converts YAML into an *Int64Value
```

File: pkg/services/provisioning/values/values.go [hunk 6]
```diff
@@ -61,15 +68,19 @@ func (val *Int64Value) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return err
 }
 
+// Value returns the wrapped int64 value
 func (val *Int64Value) Value() int64 {
 	return val.value
 }
 
+// StringValue represents a string value in a yaml
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:76]:
```suggestion
// StringValue represents a string value in a YAML
```

File: pkg/services/provisioning/values/values.go [hunk 7]
```diff
@@ -61,15 +68,19 @@ func (val *Int64Value) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
+// config that can be overridden by environment variables
 type StringValue struct {
 	value string
 	Raw   string
 }
 
+// UnmarshalYAML converts yaml into an *StringValue
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:83]:
```suggestion
// UnmarshalYAML converts YAML into an *StringValue
```

File: pkg/services/provisioning/values/values.go [hunk 8]
```diff
@@ -80,15 +91,19 @@ func (val *StringValue) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return err
 }
 
+// Value returns the wrapped string value
 func (val *StringValue) Value() string {
 	return val.value
 }
 
+// BoolValue represents a string value in a yaml
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:99]:
```suggestion
// BoolValue represents a string value in a YAML
```

File: pkg/services/provisioning/values/values.go [hunk 9]
```diff
@@ -80,15 +91,19 @@ func (val *StringValue) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
+// config that can be overridden by environment variables
 type BoolValue struct {
 	value bool
 	Raw   string
 }
 
+// UnmarshalYAML converts yaml into an *BoolValue
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:106]:
```suggestion
// UnmarshalYAML converts YAML into an *BoolValue
```

File: pkg/services/provisioning/values/values.go [hunk 10]
```diff
@@ -99,15 +114,19 @@ func (val *BoolValue) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return err
 }
 
+// Value returns the wrapped bool value
 func (val *BoolValue) Value() bool {
 	return val.value
 }
 
+// JSONValue represents a string value in a yaml
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:122]:
```suggestion
// JSONValue represents a string value in a YAML
```

File: pkg/services/provisioning/values/values.go [hunk 11]
```diff
@@ -125,15 +144,19 @@ func (val *JSONValue) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
 }
 
+// StringMapValue represents a string value in a yaml
+// config that can be overridden by environment variables
 type StringMapValue struct {
 	value map[string]string
 	Raw   map[string]string
 }
 
+// UnmarshalYAML converts yaml into an *StringMapValue
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:159]:
```suggestion
// UnmarshalYAML converts YAML into an *StringMapValue
```

File: pkg/services/provisioning/values/values.go [hunk 12]
```diff
@@ -125,15 +144,19 @@ func (val *JSONValue) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return err
 }
 
+// Value returns the wrapped json value as map[string]interface{}
 func (val *JSONValue) Value() map[string]interface{} {
 	return val.value
 }
 
+// StringMapValue represents a string value in a yaml
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:152]:
```suggestion
// StringMapValue represents a string value in a YAML
```

File: pkg/services/provisioning/values/values.go [hunk 13]
```diff
@@ -125,15 +144,19 @@ func (val *JSONValue) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return err
 }
 
+// Value returns the wrapped json value as map[string]interface{}
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:147]:
```suggestion
// Value returns the wrapped JSON value as map[string]interface{}
```

File: pkg/services/provisioning/values/values.go [hunk 14]
```diff
@@ -99,15 +114,19 @@ func (val *BoolValue) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
 }
 
+// JSONValue represents a string value in a yaml
+// config that can be overridden by environment variables
 type JSONValue struct {
 	value map[string]interface{}
 	Raw   map[string]interface{}
 }
 
+// UnmarshalYAML converts yaml into an *JSONValue
```
Comment (aknuds1) [inline pkg/services/provisioning/values/values.go:129]:
```suggestion
// UnmarshalYAML converts YAML into an *JSONValue
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Spotted a typo + left suggestions for docs improvements.

File: pkg/services/provisioning/values/values.go (hunk 3 above)
Comment (bergquist) [reply pkg/services/provisioning/values/values.go:29]: I dont see the diff >.>

Comment (aknuds1) [reply pkg/services/provisioning/values/values.go:29]: @bergquist Sorry, fixed.

**Review by aknuds1 — APPROVED** [review]

**Pull request merged**